### PR TITLE
Rebase py3 branch on top of dev

### PR DESCRIPTION
**Pull request by jameshilliard**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -72,7 +72,6 @@ if BUILD_DARWIN
 else
 	cp cppForSwig/.libs/libCppBlockUtils.so ./_CppBlockUtils.so
 endif
-endif
```
Comment (droark) [inline Makefile.am]: I believe this is still required.

File: Makefile.am [hunk 2]
```diff
@@ -144,7 +145,6 @@ if BUILD_CLIENT
 	sed "s: /usr: $(prefix):g" < dpkgfiles/armory > $(DESTDIR)$(prefix)/bin/armory
 	chmod +x $(DESTDIR)$(prefix)/bin/armory
 endif
-endif
```
Comment (droark) [inline Makefile.am]: Ditto.

File: Makefile.am [hunk 3]
```diff
@@ -87,11 +86,13 @@ install-exec-hook: uninstall-old
 if BUILD_CLIENT
 	mkdir -p $(DESTDIR)$(prefix)/lib/armory/ui
 	mkdir -p $(DESTDIR)$(prefix)/lib/armory/armoryengine
-	mkdir -p $(DESTDIR)$(prefix)/lib/armory/lang
 	mkdir -p $(DESTDIR)$(prefix)/share/applications
 	cp *.py *.so README.md $(DESTDIR)$(prefix)/lib/armory
-	cp -r ui/* $(DESTDIR)$(prefix)/lib/armory/ui
 	cp -r armoryengine/* $(DESTDIR)$(prefix)/lib/armory/armoryengine
+if HAVE_GUI
```
Comment (droark) [inline Makefile.am]: This needs an endif.

File: cppForSwig/cryptopp/rng.cpp [hunk 4]
```diff
@@ -59,25 +59,33 @@ void LC_RNG::GenerateBlock(byte *output, size_t size)
 #ifndef CRYPTOPP_IMPORTS
 
 X917RNG::X917RNG(BlockTransformation *c, const byte *seed, const byte *deterministicTimeVector)
-	: cipher(c),
-	  S(cipher->BlockSize()),
-	  dtbuf(S),
-	  randseed(seed, S),
-	  m_lastBlock(S),
-	  m_deterministicTimeVector(deterministicTimeVector, deterministicTimeVector ? S : 0)
+	: m_cipher(c),
```
Comment (droark) [inline cppForSwig/cryptopp/rng.cpp]: @goatpig will need to comment on whether or not we want to touch Crypto++. It's not used by default anymore.

**Comment by droark** [comment]
Very nice! Thanks for keeping the upgrade moving forward. It's greatly appreciated.

I looked over everything. I think the Autotools files need a few tweaks (see my comments), but once that's done and I build everything, I think I'll be happy.

File: cppForSwig/cryptopp/rng.cpp (hunk 4 above)
Comment (jameshilliard) [reply cppForSwig/cryptopp/rng.cpp]: This was from #312.

**Comment by jameshilliard** [comment]
@droark Fixed up the autotools issues, see anything I missed?

File: configure.ac [hunk 5]
```diff
@@ -182,7 +186,7 @@ case $host in
   *linux*)
     BUILD_OS=linux
 
-    if test "x$build_client" = "xyes"; then
+    if test "x$build_clientx$with_gui" = "xyesxyes"; then
```
Comment (droark) [inline configure.ac]: Is this valid? I've never seen this format before. Kinda cool - albeit a bit confusing - if it is.

**Comment by droark** [comment]
@jameshilliard - Thanks again! The Python 3 upgrade is especially critical for Macs, where it looks like whatever comes after 10.14 simply won't allow Armory to run due to Qt 4 being 32-bit.

**Pull request merged**